client/jetpack-connect/plans.jsx

```diff
@@ -297,8 +293,7 @@ export default connect(
 		const selectedSite = getSelectedSite( state );
 		const selectedSiteSlug = selectedSite ? selectedSite.slug : '*';
 
-		const selectedPlan =
-			getSiteSelectedPlan( state, selectedSiteSlug ) || getGlobalSelectedPlan( state );
+		const selectedPlan = getSiteSelectedPlan( state, selectedSiteSlug ) || retrievePlan();
```

Comment:
`getSiteSelectedPlan`:

https://github.com/Automattic/wp-calypso/blob/a9e83ba1eef06a72abee492662336d59a1361b11/client/state/jetpack-connect/selectors.js#L143-L148

https://github.com/Automattic/wp-calypso/blob/a9e83ba1eef06a72abee492662336d59a1361b11/client/state/jetpack-connect/reducer/jetpack-connect-selected-plans.js#L8-L16

There is no way for a slug to be set, can you confirm that? Let's _rip it out!_